src/datepicker/datepicker.js

```diff
@@ -629,7 +629,16 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toISOString().split('T')[0].split('-');
+                date = new Date(+dateArray[0], +dateArray[1] - 1, +dateArray[2]);
```

Comment:
After this, do `date.setFullYear(+dateArray[0])` - this is due to surprising behavior where `new Date(year)` will treat years between 0 and 100 as 19xx, such as `new Date(94)` being the year 1994 instead of 94.